reader.go

```diff
@@ -110,7 +87,7 @@ type readerImpl struct {
 	closed atomic.Bool
 
 	// TODO: Add simple LRU cache.
-	cachedFrame cachedFrame
+	cachedFrames *lru.Cache[uint64, []byte]
```

Comment:
you want to make this an interface and add an option for the reader, where you can pass the implementation of this interface.

You would also need to move single frame caching to this interface and make it a default.

PS. ARC/Two queue is probably better.